### PR TITLE
Rewrite the app to make it compatible with non-rails apps

### PR DESCRIPTION
**Pull request by dmathieu**
This is a WIP to rewrite the app and make it usable with non-rails apps.
This idea is to have a rack middleware, which we add on top of the stack and adds us a `biceps.versions` environment parameter which we can use later on.

The rails routing constraint would then use that env parameter to route or not.
And any other rack app can use that same parameter to do it's own routing.

### CODE REVIEW

**Comment by dmathieu** [comment]
I've only implemented the rack middleware for now. It's integration with the rails engine still needs to be done.

@franckverrot: WDYT

File: lib/biceps/rack.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def versions_matcher(accept)
+      accept.scan(regex)
+    end
+
+    def regex
+      Regexp.new("application/vnd.#{app_name};ver=([[:alnum:]]+)")
+    end
+
+    def app_name
+      "biceps"
```
Comment (dmathieu) [inline lib/biceps/rack.rb]: We need to find a way to nicely define the app's name (the former way wasn't cool anyway, was we are forcing the app's module name).

**Comment by dmathieu** [comment]
It works and should soon be mergeable.
I'm just thinking we could also change the app's name, to allow us to have multiple app names throughout an application.

**Comment by franckverrot** [comment]
:+1:

**Pull request merged**